doc/lua_api.txt

```diff
@@ -3303,21 +3299,67 @@ The following functions provide escape sequences:
 
 Spatial Vectors
 ===============
-A spatial vector is similar to a position, but instead using
-absolute world coordinates, it uses *relative* coordinates, relative to
-no particular point.
-
-Internally, it is implemented as a table with the 3 fields
-`x`, `y` and `z`. Example: `{x = 0, y = 1, z = 0}`.
-However, one should *never* create a vector manually as above, such misbehavior
-is deprecated. The vector helpers set a metatable for the created vectors which
-allows indexing with numbers, calling functions directly on vectors and using
-operators (like `+`). Furthermore, the internal implementation might change in
+
+Minetest stores 3-dimensional spatial vectors in Lua as tuples of 3 coordinates,
+and has a class to represent them (`vector.*`), which this chapter is about.
+For details on what a spatial vectors is, please refer to Wikipedia:
+https://en.wikipedia.org/wiki/Euclidean_vector.
+
+Spatial vectors are used for various things, including, but not limited to:
+* positions
+* position differences
+* velocities
+* euler angles (Yes, using spatial vectors here has no real semantic meaning. It's
+  just 3 values.)
+* dimensions (Sizes in 3 directions.)
+
+Note that they are *not* used for:
+* n-dimensional vectors where n is not 3 (ie. n=2)
+* arrays of the form `{num, num, num}`
+
+The API documentation may refer to spatial vectors by any of the following notations:
+* `(x, y, z)`
+* `vector.new(x, y, z)`
+* `{x=num, y=num, z=num}`
+They are all (if not stated otherwise) referring to the same thing, that is: vectors
+as produced by `vector.new`.
+
+Compatibility notes
+-------------------
+
+Internally, a vector is implemented as a table with the 3 fields `x`, `y` and `z`.
+Example: `{x = 0, y = 1, z = 0}`.
```

Comment:
IMHO this perpetuates some of the confusion.  AFAIU the new vectors implementation uses metatables, which sets them apart from old plain {x,y,z} tables.  In the way you word this here, there is no apparent distinction between the new tables and the old tables.  

Could you somehow clarify that the new tables are more special than the old tables - without confusing the reader too much with gory details of the implementation?

Reply:
You're right. Is it better now?

Reply:
Yes, very much better now.